xhr.bs

```diff
@@ -699,66 +699,72 @@ method must run these steps:
  <code>HEAD</code>, set <var>body</var> to null.
 
  <li>
-  <p>If <var>body</var> is null, go to the next step.
+  <p>If <var>body</var> is not null, then:
 
-  <p>Otherwise, let <var>encoding</var> and <var>mimeType</var> be null, and then follow these
-  rules, switching on <var>body</var>:
+  <ol>
+   <li><p>Let <var>extractedMIMEType</var> be null.
 
-  <dl class=switch>
-   <dt id=dom-xmlhttprequest-send-document>{{Document}}
-   <dd>
-    <p>Set <var>encoding</var> to `<code>UTF-8</code>`.
+   <li><p>If <var>body</var> is a {{Document}}, then set <a>request body</a> to
+   <var>body</var>, <a lt="fragment serializing algorithm">serialized</a>,
+   <a lt="obtain Unicode">converted to Unicode</a>, and <a lt="UTF-8 encode">UTF-8 encoded</a>.
 
-    <p>Set <var>mimeType</var> to `<code>text/html</code>` if <var>body</var> is an
-    <a>HTML document</a>, and to `<code>application/xml</code>` otherwise. Then append
-    `<code>;charset=UTF-8</code>` to <var>mimeType</var>.
+   <li><p>Otherwise, set <a>request body</a> and <var>extractedMIMEType</var> to the result of
+   <a for=BodyInit>extracting</a> <var>body</var>.
 
-    <p>Set <a>request body</a> to <var>body</var>,
-    <a lt="fragment serializing algorithm">serialized</a>,
-    <a lt="obtain Unicode">converted to Unicode</a>, and <a lt="UTF-8 encode">UTF-8 encoded</a>.
+   <li>
+    <p>If <a>author request headers</a> <a for="header list">contains</a>
+    `<code>Content-Type</code>`, then:
 
-   <dt id=dom-xmlhttprequest-send-bodyinit>{{BodyInit}}
-   <dd>
-    <p>If <var>body</var> is a string, set <var>encoding</var> to
-    `<code>UTF-8</code>`.
+    <ol>
+     <li>
+      <p>If <var>body</var> is a {{Document}} or a {{USVString}}, then:
 
-    <p>Set <a>request body</a> and <var>mimeType</var> to the result of
-    <a for=BodyInit>extracting</a> <var>body</var>.
-  </dl>
+      <ol>
+       <li><p>Let <var>originalAuthorContentType</var> be the <a for=header>value</a> of the
+       <a>header</a> whose <a for=header>name</a> is a <a>byte-case-insensitive</a> match for
+       `<code>Content-Type</code>` in <a>author request headers</a>.
+       <!-- XXX: add a primitive for this in Fetch -->
 
-  <p>If <var>mimeType</var> is non-null and <a>author request headers</a>
-  <a for="header list">does not contain</a> `<code>Content-Type</code>`, then
-  <a for="header list">append</a> `<code>Content-Type</code>`/<var>mimeType</var> to
-  <a>author request headers</a>.
+       <li><p>Let <var>contentTypeRecord</var> be the result of
```

Comment:
Why rename this to contentTypeRecord?

Reply:
I wanted to make it clear that this is a record, since we have both _originalAuthorContentType_ (byte sequence) and _newContentTypeSerialized_ (byte sequence). And given that it's an intermediate between those two, switching from content-type to MIME type terms was a bit confusing.

Reply:
But there's also extractedMIMEType. I think calling it MIME type throughout would be less confusing, since that's what `Content-Type` takes as value.

Reply:
extractedMIMEType isn't going to be used as a content-type. I think having to know both the field name and its type is confusing, compared to just referring to the field name consistently.

Reply:
You might want to check that again, it's used exclusively for that. I guess I'd be okay calling them all after `Content-Type`.

Reply:
Please just do whatever it is you can do to get me some help from an FBI investigator! This hacking has gotten worse since it started in 2011. I cant help y’all to do anything because I don’t know anything about coding and I just know it’s gotton so bad with these hackers that they have followed me around and they have a third party remote device and they can run my devices as if they have them in front of them. I have a lot of info I took screen shots of. I need you to contact the FBI because they have done things to harm me and even crossed state lines to do it. No matter where I go to try to file a report through the internet they are already in it because they track me. PLEASE! Help me!! They are going to end up. Killing me!!!

Reply:
I don’t know what to do with the info I read when you tell me to text in a certain form. I have no idea what I’m doing so just do what you need to do please!